src/common/copilot/PowerPagesCopilot.ts

```diff
@@ -5,27 +5,93 @@
 
 
 import * as vscode from "vscode";
-import { CodiconStylePathSegments, CopilotDisclaimer, CopilotStylePathSegments } from "./Constants";
-export let apiToken: string;
-export let sessionID: string;
-export let userName: string;
-export let orgID: string;
-export let environmentName: string;
+import { sendApiRequest } from "./IntelligenceApi";
+import { intelligenceAPIAuthentication } from "../../web/client/common/authenticationProvider";
+import { v4 as uuidv4 } from 'uuid'
+import { PacInterop, PacWrapper } from "../../client/pac/PacWrapper";
+import { PacWrapperContext } from "../../client/pac/PacWrapperContext";
+import { ITelemetry } from "../../client/telemetry/ITelemetry";
+import { AuthProfileNotFound, CodiconStylePathSegments, CopilotDisclaimer, CopilotStylePathSegments, DataverseEntityNameMap, EntityFieldMap, FieldTypeMap, WebViewMessage } from "./constants";
+import { escapeDollarSign, getLastThreeParts, getNonce, getUserName, showConnectedOrgMessage, showInputBoxAndGetOrgUrl } from "./Utils";
+import { CESUserFeedback } from "./user-feedback/CESSurvey";
+import { GetAuthProfileWatchPattern } from "../../client/lib/AuthPanelView";
+import { PacActiveOrgListOutput } from "../../client/pac/PacTypes";
+import { CopyCodeToClipboardEvent, InsertCodeToEditorEvent, UserFeedbackThumbsDownEvent, UserFeedbackThumbsUpEvent, sendTelemetryEvent } from "./telemetry/copilotTelemetry";
 
+let apiToken: string;
+let userName: string;
+let orgID: string;
+let environmentName: string;
+let userID: string;
+let activeOrgUrl: string;
+export let sessionID: string;
 
 
 export class PowerPagesCopilot implements vscode.WebviewViewProvider {
   public static readonly viewType = "powerpages.copilot";
   private _view?: vscode.WebviewView;
+  private readonly _pacWrapper: PacWrapper;
+  private _extensionContext: vscode.ExtensionContext;
+  private readonly _disposables: vscode.Disposable[] = [];
+  private loginButtonRendered = false;
+  private telemetry: ITelemetry;
 
+  constructor(private readonly _extensionUri: vscode.Uri, _context: vscode.ExtensionContext, telemetry: ITelemetry, cliPath: string) {
```

Comment:
Shouldn't this be a singleton class?

Reply:
Making it a singleton may not be a desirable choice, because it will always be instantiated in extension Activation () and will be disposed by extension on close. Singleton will introduce a global state which may open some maintainability issues.